src/reconcile/rec_write.c

```diff
@@ -2439,6 +2468,8 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 
     /* Reset the reconciliation state. */
     mod->rec_result = 0;
+    __rec_page_modify_ta_safe_free(session, &mod->ta);
```

Comment:
This deserves a comment. I thought it wasn't necessary, until I realized that checkpoint will cause this to be freed via a reconciliation without having the WT_REF locked.

Reply:
I added a comment, please check it.